app/controllers/concerns/dateable.rb

```diff
@@ -6,17 +6,22 @@ module Dateable
   private
 
   def set_dates
-    @start_date = case params[:date]
+    dateRange = params[:date] ? CGI.escapeHTML(params[:date]) : "7d"
+    @start_date = case dateRange
                   when "30d"
                     29.days.ago
                   when "this_m"
                     Time.zone.now.beginning_of_month
                   when "6m"
                     6.months.ago
+                  when "3m"
+                    3.months.ago
                   when "12m"
                     12.months.ago
-                  else
+                  when "7d"
                     6.days.ago
+                  else
+                    Time.zone.now
                   end
```

Comment:
Why not just default to `6.days.ago`and get rid of `dateRange` like before? (also don't forget to use `camel_case` in Ruby, what do you think it is? JavaScript? 😄)

Reply:
I thought it might be clearer to set the default in the beginning instead of it just being in the else statement, it looked to me a bit like an mistake. but I'll change it